### PR TITLE
Feature/fork/import

### PR DESCRIPTION
**Pull request by madhavmalhotra-3089**
Updated PR

### CODE REVIEW

File: lib/glific/contacts/import.ex [hunk 1]
```diff
@@ -3,31 +3,76 @@ defmodule Glific.Contacts.Import do
...
-             contact_id: contact.id,
-             group_id: group_id,
-             organization_id: contact.organization_id
-           }) do
-      contact
-    else
-      {:error, error} -> %{phone: contact.phone, error: error}
+  @spec process_data(map(), non_neg_integer) :: Contact.t()
+  defp process_data(%{delete: "1"} = contact, _) do
+    case Contacts.list_contacts(%{filter: %{phone: contact.phone}}) do
```
Comment (dlobo) [inline lib/glific/contacts/import.ex]: Repo.get_by(Contact, %{phone: contact.phone}) (check syntax)

{:ok, contact} ->

File: lib/glific/contacts/import.ex [hunk 2]
```diff
@@ -37,18 +82,27 @@ defmodule Glific.Contacts.Import do
   The method takes in a csv file path and adds the contacts to the particular organization
   and group.
   """
-  @spec import_contacts(String.t(), integer, integer) :: tuple()
-  def import_contacts(file_path, organization_id, group_id) do
+  @spec import_contacts(integer, String.t(), []) :: tuple()
+  def import_contacts(organization_id, group_label, opts \\ []) do
+    {date_format, opts} = Keyword.pop(opts, :date_format, nil)
```
Comment (dlobo) [inline lib/glific/contacts/import.ex]: {date_format, opts} = Keyword.pop(opts, :date_format, "{YYYY}-{....."

File: lib/glific/contacts/import.ex [hunk 3]
```diff
@@ -37,18 +86,26 @@ defmodule Glific.Contacts.Import do
...
+  def import_contacts(organization_id, group_label, opts \\ []) do
+    {date_format, opts} = Keyword.pop(opts, :date_format, "{YYYY}-{M}-{D}")
+
+    if length(opts) > 1 do
+      raise "Please specify only one of keyword arguments: file_path, url or data"
+    end
+
+    contact_data_as_stream = fetch_contact_data_as_string(opts)
+
     with %{id: organization_id} <- Partners.organization(organization_id),
```
Comment (dlobo) [inline lib/glific/contacts/import.ex]: why do u need this?

File: lib/glific/contacts/import.ex [hunk 4]
```diff
@@ -37,18 +86,26 @@ defmodule Glific.Contacts.Import do
...
-        |> File.stream!()
+         {_, group} <- Groups.get_or_create_group_by_label(group_label, organization_id) do
+      contact_data =
+        contact_data_as_stream
         |> CSV.decode(headers: true, strip_fields: true)
-        |> Enum.map(fn {_, data} -> cleanup_contact_data(data, organization_id) end)
-        |> Enum.map(fn contact -> insert_or_update_contact_data(contact, group_id) end)
+        |> Enum.map(fn {_, data} -> cleanup_contact_data(data, organization_id, date_format) end)
+
+      result =
```
Comment (dlobo) [inline lib/glific/contacts/import.ex]: You can combine thse two piped sequences into one longer sequence

File: lib/glific/groups.ex [hunk 5]
```diff
@@ -110,6 +110,23 @@ defmodule Glific.Groups do
     |> Repo.one!()
   end
 
+  @doc """
+  Get group by group name.
+  Create the group if it does not exist
+  """
+  @spec get_or_create_group_by_label(String.t(), non_neg_integer) :: {:ok, Group.t()} | nil
+  def get_or_create_group_by_label(label, organization_id) do
+    result =
```
Comment (dlobo) [inline lib/glific/groups.ex:119]: use Repo.get_by here

File: lib/glific/contacts/import.ex (hunk 3 above)
Comment (madhavmalhotra-3089) [reply lib/glific/contacts/import.ex]: Yup updating this!

File: lib/glific/groups.ex (hunk 5 above)
Comment (madhavmalhotra-3089) [reply lib/glific/groups.ex:119]: Gotcha!

**Pull request merged**